ZuluContent/Zulu/Spells/Earth/ShapeshiftSpell.cs

```diff
@@ -1,251 +1,116 @@
-            }
-        }
+            if (bodyId <= 0)
+                return;
 
-        private class InternalTimer : Timer
-        {
-            private Mobile m_Owner;
-
-            public InternalTimer(Mobile owner) : base(TimeSpan.FromSeconds(0))
+            if (!Caster.ShilCheckSkill(SkillName.Meditation, BodyData[bodyId].Difficulty, BodyData[bodyId].Points))
             {
-                m_Owner = owner;
-
-                var val = (int) owner.Skills[SkillName.Meditation].Value;
-
-                val *= 5;
-
-                if (val > 300)
-                    val = 300;
-
-                Delay = TimeSpan.FromSeconds(val);
-                Priority = TimerPriority.OneSecond;
+                Caster.SendFailureMessage("You fail to transform yourself.");
+                DoFizzle();
+                return;
             }
 
-            protected override void OnTick()
-            {
-                EndShapeshift(m_Owner);
-            }
-        }
-
-        public class ShapeshiftGump : Gump
-        {
-            private Mobile m_Caster;
-            private Item m_Scroll;
-            private Entry[] m_Entries;
-
-            public ShapeshiftGump(Mobile caster, Item scroll, Entry[] entries) : base(50, 50)
-            {
-                m_Caster = caster;
-                m_Scroll = scroll;
-                m_Entries = entries;
-
-                int x, y;
-                AddPage(0);
-                AddBackground(0, 0, 400, 550, 5054);
-                //AddBackground( 195, 36, 387, 275, 3000 );
-                AddHtmlLocalized(0, 0, 400, 18, 1015234, false, false); // <center>Polymorph Selection Menu</center>
-                AddHtmlLocalized(60, 485, 150, 18, 1011036, false, false); // OKAY
-                AddButton(25, 485, 4005, 4007, 1, GumpButtonType.Reply, 1);
-                AddHtmlLocalized(235, 485, 150, 18, 1011012, false, false); // CANCEL
-                AddButton(200, 485, 4005, 4007, 0, GumpButtonType.Reply, 2);
-
-                y = 35;
-                // TODO: This gump is probably trash, test it.
-                // yeah it's trash let's fix that
-                for (var i = 0; i < entries.Length; i++)
-                {
-                    var entry = entries[i];
-
-                    if (i > 7)
-                        x = 200;
-                    else
-                        x = 25;
-
-                    if (i == 8) y = 35;
-
-                    AddHtml(x, y, 120, 18, entry.Description, false, false);
-                    AddItem(x + 80, y, entry.BodyId);
-                    AddRadio(x, y + 20, 210, 211, false, i);
-
-                    y += 50;
-                }
-            }
-
-            public override void OnResponse(NetState state, RelayInfo info)
+            var hueMod = 0;
+            
+            Caster.FixedParticles(0x373A, 10, 10, 5007, EffectLayer.Waist);
+            Caster.PlaySound(0x209);
+            
+            Caster.TryAddBuff(new Polymorph
```

Comment:
Needs the description and maybe a different buff icon if there's one that suits?

Reply:
Ah yeah didn't think about that - I can just override the description with this { } syntax